groups/bsl/bslstl/bslstl_optional.h

```diff
@@ -2759,60 +2659,67 @@ class optional {
-
 #endif  // BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS
 
 #ifdef BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS
     template <class ANY_TYPE>
-    TYPE value_or(ANY_TYPE&& rhs) &&;
-        // If 'this->has_value() == true', return a copy of the contained
-        // 'value_type' object.  Otherwise, return the value of specified 'rhs'
-        // converted to 'value_type'.
-
+    TYPE value_or(ANY_TYPE&& value) &&;
+        // Return a copy of the underlying object of a (template parameter)
+        // 'TYPE' if this object is non-null, and the specified 'value'
+        // converted to 'TYPE' otherwise.  Note that this method returns *by*
+        // *value*, so may be inefficient in some contexts.
 #ifdef BSL_COMPILERFEATURES_GUARANTEED_COPY_ELISION
     template <class ANY_TYPE>
-    TYPE value_or(bsl::allocator_arg_t, allocator_type, ANY_TYPE&&) &&;
-      // If 'this->has_value() == true', return a copy of the contained
-      // 'value_type' object created using the provided allocator.  Otherwise,
-      // return the value of specified 'rhs' converted to 'value_type' and use
-      // the specified 'allocator' for the returned object.
-#endif  //BSL_COMPILERFEATURES_GUARANTEED_COPY_ELISION
-#endif  //BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS
+    TYPE value_or(bsl::allocator_arg_t, allocator_type, ANY_TYPE&& value) &&;
+        // If this object is non-null, return a copy of the the underlying
+        // object of a (template parameter) 'TYPE' created using the provided
+        // allocator, and the specified 'value' converted to 'TYPE' using the
+        // specified 'allocator' otherwise.  Note that this method returns *by*
+        // *value*, so may be inefficient in some contexts.
+#endif  // BSL_COMPILERFEATURES_GUARANTEED_COPY_ELISION
+#endif  // BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS
 
     optional& operator=(bsl::nullopt_t) BSLS_KEYWORD_NOEXCEPT;
         // Reset this object to be disengaged and return a reference providing
         // modifiable access to this object.
 
     optional& operator=(const optional& rhs);
+        // Assign to this object the value of the specified 'rhs' object, and
+        // return a non-'const' reference to this object.
+
     optional& operator=(BloombergLP::bslmf::MovableRef<optional> rhs);
-        // If specified 'rhs' is engaged, assign its value to this object.
-        // Otherwise, reset this object to a disengaged state.  Return a
-        // reference providing modifiable access to this object.  Note that
-        // this method may invoke assignment from 'rhs', or construction from
-        // 'rhs', depending on whether this 'optional' object is engaged.
+        // Assign to this object the value of the specified 'rhs' object, and
+        // return a non-'const' reference to this object.  The allocators of
+        // this object and 'rhs' both remain unchanged.  The contents of 'rhs'
+        // are either move-inserted into or move-assigned to this object.
```

Comment:
`NullableValue` has this doc too, but `move-inserted` doesn't make much sense in this doc to me.  If you have a better term in mind - please change it (throughout).  Otherwise, let's leave it as is.

Reply:
i think i'd go for 'move-constructed'